librocketchat.c

```diff
@@ -837,9 +874,15 @@ rc_fetch_url(RocketChatAccount *ya, const gchar *url, const gchar *postdata, Roc
 	gchar *host = NULL, *path = NULL, *user = NULL, *password = NULL;
 	int port;
 	purple_url_parse(url, &host, &port, &path, &user, &password);
-	
+
 	headers = g_string_new(NULL);
-	
+
+	if (ya->websocket_server != ya->server) {
+		host = g_strdup(ya->websocket_server);

```

Comment:
Does this leak the memory of the old `host`?  Need `g_free(host)` before assign?

Reply:
Do I need to do so? host is just a plain char array without malloc created in the purple function.